engine/lib/output.php

```diff
@@ -28,20 +28,18 @@ function parse_urls($text) {
 	// if PHP supported conditional negative lookbehinds we could use this:
 	// $r = preg_replace_callback('/(?<!=)(?<![ ])?(?<!["\'])((ht|f)tps?:\/\/[^\s\r\n\t<>"\'\!\(\),]+)/i',
 	$r = preg_replace_callback('/(?<![=\/"\'])((ht|f)tps?:\/\/[^\s\r\n\t<>"\']+)/i',
-	create_function(
-		'$matches',
-		'
-			$url = $matches[1];
-			$punc = "";
-			$last = substr($url, -1, 1);
-			if (in_array($last, array(".", "!", ",", "(", ")"))) {
-				$punc = $last;
-				$url = rtrim($url, ".!,()");
-			}
-			$urltext = str_replace("/", "/<wbr />", $url);
-			return "<a href=\"$url\" rel=\"nofollow\">$urltext</a>$punc";
-		'
-	), $text);
+	function($matches) {
+		$url = $matches[1];
+		$punc = '';
+		$last = substr($url, -1, 1);
+		if (in_array($last, array(".", "!", ",", "(", ")"))) {
+			$punc = $last;
+			$url = rtrim($url, ".!,()");
+		}
+		$urltext = str_replace("/", "/<wbr />", $url);
+		
+		return "<a href=\"$url\" rel=\"nofollow\">$urltext</a>$punc";
+	}, $text);
```

Comment:
👍 for real code but performance would be a little worse here since it creates the function object for each call. Better to extract this function to `_elgg_parse_urls_callback`.